### PR TITLE
feat(docs): add impersonation guide

### PR DESCRIPTION
**Pull request by Kosai106**
I took the liberty of also updating the guide to make use of the new resolvers.

My writing skills are not the best so I didn't add too much additional content from the original guide so feel free to come with suggestions for improvements if necessary.

### CODE REVIEW

**Comment by flybayer** [comment]
Sweet thanks!! You need to rename `impersonation.tsx` to `impersonation.mdx` :)

And then I'll review further

**Comment by Kosai106** [comment]
@flybayer you're right 🤦‍♂️ Old habits die hard.

**Review by JuanM04 — APPROVED** [review]
Very nice! I wouldn't used assert, but it works

**Comment by Kosai106** [comment]
> Very nice! I wouldn't used assert, but it works

It's a left over from the original guide although maybe it makes sense to replace `assert` and `logger` as I don't think they come with Blitz projects by default. Let me know what you think and I'll update the PR accordingly.

**Comment by JuanM04** [comment]
Yes, the paths are wrong... maybe replace them by a simple "if not exist, then throw" and a "console.log". For simplicity 😄

**Review by JuanM04 — APPROVED** [review]

File: app/pages/docs/impersonation.mdx [hunk 1]
```diff
@@ -0,0 +1,157 @@
+---
+title: How To Impersonate Other Users
+sidebar_label: How To Impersonate Other Users
+---
+
+Impersonation, also called personification is often a critical tool for SaaS applications as it allows you to view the application from the perspective of a specific user without knowing their credentials.
```
Comment (flybayer) [inline app/pages/docs/impersonation.mdx:6]:
```suggestion
Impersonation is often a critical tool for SaaS applications as it allows admins to view the application from the perspective of a specific user without knowing their credentials.
```

"personification" is something totally different :)

**Comment by flybayer** [comment]
Awesome, thanks so much!! Now we can remove it from the wiki

**Pull request merged**